juju/juju
# Prune actions (results)

## Description of change

Fixes referenced bug

Like `Status Histories`, actions (and their results) are now pruned according to a configured persistent storage size and entry age.  

## QA steps

Two new model configuration parameters where added, `max-action-age` and `max-action-size`.

Check to see if these to configuration parameters are listed as valid parameters

```sh
juju model-config
```

The pruning interval for actions is currently set to two weeks (24 * time.hour). This prohibits examining the pruning action from outside of the code. However there are several ways you can test this which include. 

- [Fake the time](https://github.com/wolfcw/libfaketime) of the machine agent for which the action is running
- Temporarily alter the code to reduce the pruning interval

Of course it would be much nicer if the interval was user configurable.

## Documentation changes

No

## Bug reference

[lp#1703047](https://bugs.launchpad.net/juju/+bug/1703047)
